common/djangoapps/third_party_auth/samlproviderconfig/views.py

```diff
@@ -0,0 +1,56 @@
+"""
+    Viewset for auth/saml/v0/samlproviderconfig
+"""
+
+from edx_rbac.mixins import PermissionRequiredMixin
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from enterprise.models import EnterpriseCustomerIdentityProvider
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from third_party_auth.samlutils.utils import fetch_enterprise_customer_by_id
+
+from ..models import SAMLProviderConfig
+from .serializers import SAMLProviderConfigSerializer
+
+
+class SAMLProviderMixin(object):
+    authentication_classes = [JwtAuthentication, SessionAuthentication]
+    permission_classes = [permissions.IsAuthenticated]
+    serializer_class = SAMLProviderConfigSerializer
+
+
+class SAMLProviderConfigViewSet(PermissionRequiredMixin, SAMLProviderMixin, viewsets.ModelViewSet):
```

Comment:
If you only want this Viewset to support the list and create operations, you could inherit from `mixins.ListModelMixin, mixins.CreateModelMixin, viewsets.GenericViewSet`.  See https://www.django-rest-framework.org/api-guide/viewsets/#example_3

Reply:
ok thanks have to check with @taliaedX  on that, I believe we also want patch .. for update etc.. but I am only testing for get and post right now ...